alibi/explainers/tests/test_integrated_gradients.py

```diff
@@ -650,6 +652,68 @@ def test_run_forward_from_layer(layer_nb,
     assert np.allclose(preds_from_layer, expected_values)
 
 
+TARGETS_ARGS = [{'output_shape': (None, 2), 'target': np.array([0, 2]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None,), 'target': np.array([0]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 3), 'target': np.array([0, 0, 0]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 3), 'target': np.array([-123, -123]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 3), 'target': np.array([99, 99, 99]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 3), 'target': 999, 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 2), 'target': np.array([[0, 2]]), 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 4, 4, 3), 'target': np.array([[0, 0], [0, 0]]),
+                 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 4, 4, 3), 'target': np.array([[0, 0, 3], [0, 0, 0]]),
+                 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 4, 4, 3), 'target': np.array([[0, 4, 0], [0, 0, 0]]),
+                 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 4, 4, 3), 'target': np.array([[99, 99, 99], [0, 0, 0]]),
+                 'nb_samples': 2, 'bad_target': True},
+                {'output_shape': (None, 2), 'target': np.array([0, 1]), 'nb_samples': 2, 'bad_target': False},
+                {'output_shape': (None,), 'target': np.array([0, 1]), 'nb_samples': 2, 'bad_target': False},
+                {'output_shape': (None, 3), 'target': np.array([0, 2]), 'nb_samples': 2, 'bad_target': False},
+                {'output_shape': (None, 4, 4, 3), 'target': np.array([[0, 0, 0], [3, 3, 2]]),
+                'nb_samples': 2, 'bad_target': False}]
+
+SELECT_ARGS = [{'preds': np.array([[0.0, 0.1],
+                                   [1.0, 1.1]]),
+                'target': np.array([0,
+                                    0]),
+                'expected': np.array([0.0,
+                                      1.0])},
+               {'preds': np.array([[[0.0, 0.1], [1.0, 1.1]],
+                                   [[2.0, 2.1], [3.0, 4.1]]]),
+                'target': np.array([[0, 0],
+                                   [0, 0]]),
+                'expected': np.array([[0.0],
+                                      [2.0]])}]
```

Comment:
I see that this test case tests a 3-dimensional output and a 2-dimensional target which partially answers my question above why 2D targets are allowed. However, I though we only support 2-dimensional outputs anyway?

Does this mean that this PR implicitly allows up to 3D output?

Reply:
yes, that's right. Any rank for the output tensor are allowed now, not only 3D. Refer to comment above